bullet-featherstone/src/SDFFeatures.cc

```diff
@@ -951,6 +1269,73 @@ Identity SDFFeatures::ConstructSdfCollision(
   return this->GenerateInvalidId();
 }
 
+/////////////////////////////////////////////////
+Identity SDFFeatures::ConstructSdfJoint(
+    const Identity &_modelID,
+    const ::sdf::Joint &_sdfJoint)
+{
+  auto modelInfo = this->ReferenceInterface<ModelInfo>(_modelID);
+  if (_sdfJoint.ChildName() == "world")
+  {
+    gzerr << "Asked to create a joint with the world as the child in model "
+           << "[" << modelInfo->name << "]. This is currently not "
+           << "supported\n";
+
+    return this->GenerateInvalidId();
+  }
+
+  std::string parentLinkName;
+  const auto resolveParentErrors = _sdfJoint.ResolveParentLink(parentLinkName);
+  if (!resolveParentErrors.empty()) {
+    parentLinkName = _sdfJoint.ParentName();
+  }
+  std::string childLinkName;
+  const auto childResolveErrors = _sdfJoint.ResolveChildLink(childLinkName);
+  if (!childResolveErrors.empty()) {
+    childLinkName = _sdfJoint.ChildName();
+  }
+
+  // Currently only supports constructing joint with world
+  if (parentLinkName == "world" && _sdfJoint.Type() == ::sdf::JointType::FIXED)
+  {
+    auto worldModelIt = this->models.find(_modelID);
+    if (worldModelIt == this->models.end())
+      return this->GenerateInvalidId();
+    const auto worldModel = worldModelIt->second;
+    std::size_t idx = childLinkName.find("::");
+    if (idx == std::string::npos)
+      return this->GenerateInvalidId();
+
+    const std::string modelName = childLinkName.substr(0, idx);
+    std::size_t modelID = worldModel->nestedModelNameToEntityId.at(modelName);
+    auto model = this->models.at(modelID);
+
+    model->body->setFixedBase(true);
```

Comment:
the `btMultiBody::setFixedBase` API was added in https://github.com/bulletphysics/bullet3/commit/19aafd522114906c0f4a35a5fcde58400e382f5d and released in bullet versions 2.89 and 3.05. Ubuntu 20.04 CI is failing as it only has bullet version 2.88 (22.04 CI looks fine). If this API is needed, we may need to retarget to Harmonic

Reply:
added `ifdef` for bullet version in 20.04. `ConstructSdfJoint` will not work in versions < 2.89. [88962e8](https://github.com/gazebosim/gz-physics/pull/574/commits/88962e84cda332c69e27fa5e83e2dfc2d1427d16)

Reply:
I've proposed using 22.04 for GitHub workflow code coverage testing in #578, which should give use us better insight into the coverage of this feature

Reply:
expanded test to cover the construct joint feature. [15aab54](https://github.com/gazebosim/gz-physics/pull/574/commits/15aab54d1e02a0b1e618b346ed19e48726ae6a6c)